src/views/forms/TradeForm/TradeSizeInputs.tsx

```diff
@@ -167,47 +174,47 @@ export const TradeSizeInputs = () => {
     );
   };
 
+  const inputConfig = {
+    [DisplayUnit.Asset]: {
+      onInput: onSizeInput,
+      type: InputType.Number,
+      tooltipId: 'order-amount',
+      value: amountInput,
+    },
+    [DisplayUnit.Fiat]: {
+      onInput: onUSDCInput,
+      type: InputType.Currency,
+      tooltipId: 'order-amount-usd',
+      value: usdAmountInput,
+    },
+  }[displayUnit];
+
   const sizeInput = (
     <FormInput
       id="trade-amount"
       decimals={decimals}
-      onInput={onSizeInput}
-      label={
-        <>
-          <WithTooltip tooltip="order-amount" stringParams={{ SYMBOL: id ?? '' }} side="right">
-            {stringGetter({ key: STRING_KEYS.AMOUNT })}
-          </WithTooltip>
-          {id && <Tag>{id}</Tag>}
-        </>
-      }
-      slotRight={inputToggleButton()}
-      type={InputType.Number}
-      value={amountInput || ''}
-    />
-  );
-
-  const usdcInput = (
-    <FormInput
-      id="trade-usdc"
-      onInput={onUSDCInput}
-      type={InputType.Currency}
-      value={usdAmountInput || ''}
-      decimals={tickSizeDecimals ?? USD_DECIMALS}
+      onInput={inputConfig.onInput}
       label={
         <>
-          <WithTooltip tooltip="order-amount-usd" stringParams={{ SYMBOL: id ?? '' }} side="right">
+          <WithTooltip
+            tooltip={inputConfig.tooltipId}
+            stringParams={{ SYMBOL: id ?? '' }}
+            side="right"
+          >
             {stringGetter({ key: STRING_KEYS.AMOUNT })}
           </WithTooltip>
-          <Tag>USD</Tag>
+          {id && <DisplayUnitTag tw="ml-0.25" assetId={id} entryPoint="tradeAmountInputAssetTag" />}
```

Comment:
nit: use 0.5ch to be consistent with `Tag`?

Reply:
honestly makes me think it would be nice to add spacing constants to `globalStyle.ts` (like `--tag-padding` or something)